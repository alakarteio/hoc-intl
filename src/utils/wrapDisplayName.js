export default (Component, wrapperName) => (
  `${wrapperName || 'intl'} (${Component.displayName
  || Component.name
  || (Component.constructor && Component.constructor.name)
  || 'Unknown'})`
)
