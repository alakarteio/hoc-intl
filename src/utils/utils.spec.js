/* eslint-env jest */
import wrapDisplayName from './wrapDisplayName'

const wrapperName = 'wrapper'

describe('utils/wrapDisplayName', () => {
  it(`should print a displayName with custom name : ${wrapperName}`, () => {
    expect({
      'Component.displayName': wrapDisplayName({ displayName: 'Component.displayName' }, wrapperName),
      'Component.name': wrapDisplayName({ name: 'Component.name' }, wrapperName),
      'Component.constructor.name': wrapDisplayName({ constructor: { name: 'Component.constructor.name' } }, wrapperName),
      Unknown: wrapDisplayName({ constructor: {} }, wrapperName),
    }).toMatchSnapshot()
  })
  it('should print a displayName', () => {
    expect({
      'Component.displayName': wrapDisplayName({ displayName: 'Component.displayName' }),
      'Component.name': wrapDisplayName({ name: 'Component.name' }),
      'Component.constructor.name': wrapDisplayName({ constructor: { name: 'Component.constructor.name' } }),
      Unknown: wrapDisplayName({ constructor: {} }),
    }).toMatchSnapshot()
  })
})
