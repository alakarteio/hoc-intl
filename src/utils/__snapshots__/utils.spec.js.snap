// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/wrapDisplayName should print a displayName 1`] = `
Object {
  "Component.constructor.name": "intl (Component.constructor.name)",
  "Component.displayName": "intl (Component.displayName)",
  "Component.name": "intl (Component.name)",
  "Unknown": "intl (Unknown)",
}
`;

exports[`utils/wrapDisplayName should print a displayName with custom name : wrapper 1`] = `
Object {
  "Component.constructor.name": "wrapper (Component.constructor.name)",
  "Component.displayName": "wrapper (Component.displayName)",
  "Component.name": "wrapper (Component.name)",
  "Unknown": "wrapper (Unknown)",
}
`;
